engine/schema/src/main/java/com/cloud/upgrade/DatabaseVersionHierarchy.java

```diff
@@ -131,7 +131,7 @@ private CloudStackVersion getRecentVersion(final CloudStackVersion fromVersion)
         return toList()
                  .reverse()
                  .stream()
-                 .filter(version -> fromVersion.compareTo(version) < 0)
```

Comment:
Should we just remove the last digit (from a.b.c.d; remove .d) in the upgrade paths map and the logic that obtains current ACS version (to report a.b.c only to DB upgrade path processors?)

Reply:
unfortunately a lot of four digit versions are already in the DBs around the world. In addition to that, we might want to add a DB patch in a security release at some pioint in which case we want to take the fourth digit into account.